### ISSUE TITLE
What's the intended functionality of gr.Image's `streaming` feature?

### ISSUE THREAD

**Issue by nateraw** [gradio-app/gradio#1490]
The [gr.Image docs](https://gradio.app/docs/#image-header) say this about the `streaming` kwarg:

> If True when used in a `live` interface, will automatically stream webcam feed. Only valid is source is 'webcam'.

Here's the minimal example I used:

```python
import gradio as gr

gr.Interface(lambda x: x, gr.Image(source='webcam', streaming=True), "image").launch()
```

When I run the following, the difference between `streaming=False` and `streaming=True` is just that I don't have to click the camera snapshot button and can just hit submit...this doesn't really feel like "streaming" to me, so I'm curious if that's the intended functionality.

Based on the description of this arg, I would have expected some live feed of images being passed through my function at some set FPS, so effectively I would be seeing an altered version of my webcam video.

**Comment by abidlabs** [comment]
Ah so to get the proper streaming effect, you also have to add `live=True`, like this:

```py
import gradio as gr

gr.Interface(lambda x: x, gr.Image(source='webcam', streaming=True), "image", live=True).launch()
```

However, the loading indicator kind of messes up the live effect. Pinging @aliabid94:
* Can we document/clarify how the `streaming` parameter should be used? Does it need to be used with `live=True`?
* Can we fix the loading indicator in this case to not to cover the entire output?

**Comment by nateraw** [comment]
Thanks for that snippet! It works now. I added an issue to track what you mentioned about the loading indicator, as that was immediately noticeably frustrating to me too.

**Comment by aliabid94** [comment]
I added `streaming=True` because live=True was ambiguous whether it should trigger when the user changes the image (e.g. takes a snapshot) or actually stream frames. In Blocks mode, this kwarg isn't necessary, because you can just add a streaming listener, e.g.
```
cam = gr.Image(source="webcam")
cam.stream(fn, cam, output)
```

**Comment by aliabid94** [comment]
Basically, `live=True` attaches a `change` listener in Interface mode unless a component has streaming=True (which Image and Audio enable), in which case it attaches a `stream` listener instead.

**Comment by abidlabs** [comment]
Good stuff to document in your guide @nateraw ;)

 I think we can close this issue, as the only remaining todo is to fix the loader, which is in #1494
